tests/testthat/test-zzz-methods-qmatrix-validation.R

```diff
@@ -0,0 +1,70 @@
+test_that("Q-matrix validation works for ecpe", {
+  # correctly specified dina model ---------------------------------------------
+  qmat_valid_res <- qmatrix_validation(x = rstn_dina)
+  expect_equal(
+    names(qmat_valid_res),
+    c(
+      "item_id",
+      "original_specification",
+      "original_pvaf",
+      "empirical_specification",
+      "empirical_pvaf"
+    )
+  )
+  expect_equal(nrow(qmat_valid_res), 35)
+  expect_equal(
+    nrow(
+      qmat_valid_res |>
+        dplyr::filter(is.na(empirical_specification))
+    ),
+    35
+  )
+
+  # misspecified dino model ----------------------------------------------------
+  qmat_valid_res <- qmatrix_validation(x = rstn_dino)
+  expect_equal(
+    names(qmat_valid_res),
+    c(
+      "item_id",
+      "original_specification",
+      "original_pvaf",
+      "empirical_specification",
+      "empirical_pvaf"
+    )
+  )
+  expect_equal(nrow(qmat_valid_res), 35)
+  expect_equal(
+    nrow(
+      qmat_valid_res |>
+        dplyr::filter(is.na(empirical_specification))
+    ),
+    35
+  )
```

Comment:
This test is currently passing, but I don't think it should. In `tests/testthat/setup.R`, I updated the DINO model intentionally use a Q-matrix that is misspecified from what was used to generate the data:

https://github.com/wjakethompson/measr/blob/5eaa11b66ccf57332dc82842d31a27344039f47b/tests/testthat/setup.R#L8-L18

But I still get no recommended changes, and the pvaf for original (and in 4 cases incorrect) Q-matrix specifications are all 1.0.

I find it highly implausible that the pvaf is always 1, for every data set (ECPE in the documentation example, DINA with correct Q-matrix and DINO with incorrect Q-matrix here). So I think there is likely a problem with how sigma is being calculated?